src/main_benchmark.cpp

```diff
@@ -218,8 +221,100 @@ void triggerInferenceInALoop(
     averageWholeLatency = std::accumulate(latenciesWhole.begin(), latenciesWhole.end(), 0) / (double(niterPerThread) * 1'000);
     averagePureLatency = std::accumulate(latenciesPure.begin(), latenciesPure.end(), 0) / (double(niterPerThread) * 1'000);
 }
+
+void triggerInferenceInALoopResetBuffer(
+    std::future<void>& startSignal,
+    std::promise<void>& readySignal,
+    const size_t niterPerThread,
+    size_t& wholeThreadTimeUs,
+    double& averageWholeLatency,
+    double& averagePureLatency,
+    OVMS_Server* server,
+    OVMS_InferenceRequest* request,
+    const std::string& inputName,
+    std::uint32_t elementsCount) {
+    OVMS_InferenceResponse* response{nullptr};
+    std::vector<uint64_t> latenciesWhole(niterPerThread);
+    std::vector<uint64_t> latenciesPure(niterPerThread);
+    std::vector<std::vector<float>> preparedData;
+    for (int i = 0; i < niterPerThread; i ++) {
+        float random = ((float) rand()) / (float) RAND_MAX;
+        std::vector<float> data(elementsCount, random);
+        preparedData.push_back(data);
+    }
+    readySignal.set_value();
+    startSignal.get();
+    auto workloadStart = std::chrono::high_resolution_clock::now();
+    size_t iter = 0;
+    while (iter < niterPerThread) {
+        auto iterationWholeStart = std::chrono::high_resolution_clock::now();
+        OVMS_InferenceRequestInputRemoveData(request, inputName.c_str());
```

Comment:
Here if request doesnt have any Input (this could be true for first iteration) we will get error and allocate ovms::Status(ret code NONEXISTENT_BUFFER_FOR_REMOVAL) and wouldn't free it. At this point does the infer request have input bufer set? If yes then this code is fine